ts/Core/Axis/GridAxis.ts

```diff
@@ -966,7 +966,7 @@ class GridAxis {
 
             // Merge the user options into default grid axis options so
             // that when a user option is set, it takes presedence.
-            gridAxisOptions = merge(true, {
+            gridAxisOptions = merge<DeepPartial<Highcharts.AxisOptions>>(true, {
```

Comment:
This should be no in-place merge. -> no `true`

Reply:
Why not? We are creating a new object, so instead of creating another object we might just as well merge the `userOptions` into our local defaults?

Reply:
You are right.